### PR TITLE
timers inside sandbox

### PR DESCRIPTION
**Pull request by shamasis**
Please do a deep review to let me know if any test case is missing (other than very similar ones)

### CODE REVIEW

File: lib/sandbox/execute-context.js [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+    legacy = require('./postman-legacy-interface'),
+
+    NONLEGACY_SANDBOX_MARKERS = {
+        '"use sandbox2";': true,
+        '\'use sandbox2\';': true
+    };
+
+module.exports = function (scope, code, execution, console, timers, pmapi) {
+    // if there is no code, then no point bubbling anything up
+    if (!(code && _.isString(code))) {
```
Comment (kunagpal) [inline lib/sandbox/execute-context.js]: Can be replaced with:
```js
if (!_.isString(code)) {
```

File: lib/sandbox/postman-legacy-interface.js [hunk 2]
```diff
@@ -317,7 +320,7 @@ module.exports = {
         }
 
         // 5. add the iteration information
-        globalvars.iteration = _.isObject(options.cursor) ? options.cursor.iteration : 0;
+        globalvars.iteration = _.isObject(execution.cursor) ? execution.cursor.iteration : 0;
```
Comment (kunagpal) [inline lib/sandbox/postman-legacy-interface.js]: Can be replaced with `_.get(execution.cursor, 'iteration', 0);`

File: lib/sandbox/timers.js [hunk 3]
```diff
@@ -0,0 +1,273 @@
...
+    timerFunctionNames = ['Timeout', 'Interval', 'Immediate', 'Event'],
+
+    /**
+     * This object defines a set of timer function names that are trigerred a number of times instead of a single time.
+     * Such timers, when placed in generic rules, needs special attention.
+     * @constant {Array.<Boolean>}
+     */
+    multiFireTimerFunctions = {
+        'Interval': true
+
```
Comment (kunagpal) [inline lib/sandbox/timers.js]: Extra blank line.

File: lib/sandbox/timers.js [hunk 4]
```diff
@@ -0,0 +1,273 @@
...
+        dummyContext = {},
+
+        total = 0, // accumulator to keep track of total timers
+        pending = 0, // counters to keep track of running timers
+        sealed = false, // flag that stops all new timer additions
+        wentAsync = false,
+        computeTimerEvents;
+
+    // do special handling to enable emulation of immediate timers in hosts that lacks them
+    if (typeof timers.setImmediate !== 'function') {
```
Comment (kunagpal) [inline lib/sandbox/timers.js]: `'function'` can be referenced as `FUNCTION`

File: lib/sandbox/timers.js [hunk 5]
```diff
@@ -0,0 +1,273 @@
...
+        else {
+            this[('clearAll' + name + 's')] = function () {
+                Object.keys(running).forEach(function (id) {
+                    computeTimerEvents(-1, true);
+                    // run clear functions except for static timers
+                    timers['clear' + name](running[id]);
+                });
+            };
+        }
+
```
Comment (kunagpal) [inline lib/sandbox/timers.js]: Extra blank lines

File: lib/sandbox/execute-context.js (hunk 1 above)
Comment (shamasis) [inline lib/sandbox/execute-context.js]: no. empty string should not be allowed too.

**Pull request merged**